### PR TITLE
docs: document required capabilities

### PR DESCRIPTION
**Pull request by swiatekm-sumo**
##### Description

Document what capabilities collection requires to function for each data type and how some of the default permissions may be restricted.

---

##### Checklist

Remove items which don't apply to your PR.

- [x] Changelog updated

### CODE REVIEW

File: deploy/docs/Security_Best_Practices.md [hunk 1]
```diff
@@ -2,6 +2,59 @@
...
+As Kubernetes distributions have different ACLs set for these directories, Fluent-Bit runs as the root user by default
+for maximum out-of-the-box compatibility. This can be set to any user or group id with permission to read log files
+in the aforementioned directories on the node. In terms of collection configuration, the value to modify is
+`fluent-bit.securityContext` for Fluent-Bit.
+
+The log collector needs to be able to talk to the log metadata enrichment service. It does not need to do any other
+requests over the network.
+
+The log metadata enrichment service needs to be able to talk to the Sumo Logic backend receiver endpoints. It also needs
+to be able to access the Kubernetes API Server to pull metadata.
```
Comment (pmalek-sumo) [inline deploy/docs/Security_Best_Practices.md:40]: > to pull metadata

Or to get notified via watch when user is using otelcol for metadata enrichment.

File: deploy/docs/Security_Best_Practices.md [hunk 2]
```diff
@@ -2,6 +2,59 @@
...
+
+Pipelines for the three different data types collection supports - logs, metrics and traces - are completely
+independent in this respect, so we're going to discuss each one separately. If a particular data type is
+not enabled in the configuration, there is no need to do anything, as the respective components will
+simply not be started.
+
+Logically, the pipeline for a data type is composed of the following components
+
+```
+Collector -> (Optional) Metadata Enrichment -> Sumologic Backend
```
[the remark this answers is not in the document]
Comment (swiatekm-sumo) [reply deploy/docs/Security_Best_Practices.md:18]: For traces the collector and metadata enrichment are the same thing. Maybe there's a better way of expressing that?

File: deploy/docs/Security_Best_Practices.md [hunk 3]
```diff
@@ -2,6 +2,59 @@
...
+\- `/var/log/containers` for container logs and `/var/log/journal` for journald logs - mounted as volumes in the
+agent container. In order to read them, the agent container needs to run as a user with the right permissions.
+
+As Kubernetes distributions have different ACLs set for these directories, Fluent-Bit runs as the root user by default
+for maximum out-of-the-box compatibility. This can be set to any user or group id with permission to read log files
+in the aforementioned directories on the node. In terms of collection configuration, the value to modify is
+`fluent-bit.securityContext` for Fluent-Bit.
+
+The log collector needs to be able to talk to the log metadata enrichment service. It does not need to do any other
+requests over the network.
```
[the remark this answers is not in the document]
Comment (swiatekm-sumo) [reply deploy/docs/Security_Best_Practices.md:37]: Yes, that's exactly what the passage you quoted says. Am I misunderstanding something?

File: deploy/docs/Security_Best_Practices.md (hunk 1 above)
Comment (swiatekm-sumo) [reply deploy/docs/Security_Best_Practices.md:40]: This is intended as a high-level description. It pulls the data via a WATCH.

Comment (pmalek-sumo) [reply deploy/docs/Security_Best_Practices.md:40]: It doesn't technically pull it. The notifications get pushed to the client AFAIK.

File: deploy/docs/Security_Best_Practices.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (pmalek-sumo) [reply deploy/docs/Security_Best_Practices.md:37]: > ... not ...

yes 🤦

File: deploy/docs/Security_Best_Practices.md (hunk 1 above)
Comment (pmalek-sumo) [reply deploy/docs/Security_Best_Practices.md:40]: obtain, receive, get 🤷

Comment (swiatekm-sumo) [reply deploy/docs/Security_Best_Practices.md:40]: I like `obtain`.

File: deploy/docs/Security_Best_Practices.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (swiatekm-sumo) [reply deploy/docs/Security_Best_Practices.md]: Removed the `Optional`, the description for traces makes this clear enough imo.

File: CHANGELOG.md [hunk 4]
```diff
@@ -11,6 +11,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Changed
 
+- docs: document required capabilities [#2051][#2051]
```
Comment (pmalek-sumo) [inline CHANGELOG.md:14]: I don't believe this change should land in changelog

**Review by pmalek-sumo — APPROVED** [review]

**Pull request merged**